packages/sales-api-service/src/services/transactions/process-transaction-queue.js

```diff
@@ -68,6 +65,9 @@ export async function processQueue ({ id }) {
       isRenewal
     )
 
+    const { recurringPayment } = await processRecurringPayment(transactionRecord, contact)
+    recurringPayment && entities.push(recurringPayment)
```

Comment:
You should be testing for the call out to processRecurringPayment, that entities has the return value of processRecurringPayment pushed on when it returns a value and that it doesn't when processRecurringPayment returns a null or undefined.

Also, please don't use logical value operators in this way - it's ok for assigning values to a variable, but it's clearer to use an if statement in this case:
```
if (recurringPayment) {
  entities.push(recurringPayment)
}
```